Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -39,14 +39,15 @@ class ChannelUpdater: Worker {
                 let payload = try result.get()
                 channelCreatedCallback?(payload.channel.cid)
                 database?.write { session in
-                    let channelDTO = session.channel(cid: payload.channel.cid)
-                    channelDTO?.cleanMessagesThatFailedToBeEditedDueToModeration()
-
-                    if isFirstPage, let channelDTO = channelDTO {
+                    if let channelDTO = session.channel(cid: payload.channel.cid) {
+                        channelDTO.cleanMessagesThatFailedToBeEditedDueToModeration()
+                    }
+                    
+                    if isFirstPage, let channelDTO = session.channel(cid: payload.channel.cid) {
                         channelDTO.messages = channelDTO.messages.filter { $0.localMessageState?.isLocalOnly == true }
                     }
 
-                    try session.saveChannel(payload: payload, isPaginatedPayload: !isFirstPage)
+                    try session.saveChannel(payload: payload)
```

Comment:
We can actually reset the `oldestMessageAt` here and it will also fix the problem 🤔 Not requiring the fetchLimit and batchSize